Modules/Microsoft365DSC/DSCResources/MSFT_PPTenantIsolationSettings/MSFT_PPTenantIsolationSettings.psm1

```diff
@@ -0,0 +1,688 @@
+function Get-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.Collections.Hashtable])]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [ValidateSet('Yes')]
+        [System.String]
+        $IsSingleInstance,
+
+        [Parameter()]
+        [System.Boolean]
+        $Enabled = $true,
+
+        [Parameter()]
+        [Microsoft.Management.Infrastructure.CimInstance[]]
+        $Rules,
+
+        [Parameter()]
+        [Microsoft.Management.Infrastructure.CimInstance[]]
+        $RulesToInclude,
+
+        [Parameter()]
+        [Microsoft.Management.Infrastructure.CimInstance[]]
+        $RulesToExclude,
+
+        [Parameter()]
+        [System.Management.Automation.PSCredential]
+        $Credential
+    )
+
+    Write-Verbose -Message 'Getting the Power Platform Tenant Isolation Settings Configuration'
+
+    if ($PSBoundParameters.ContainsKey("Rules") -and `
+        ($PSBoundParameters.ContainsKey("RulesToInclude") -or `
+                $PSBoundParameters.ContainsKey("RulesToExclude")))
+    {
+        $message = 'You cannot specify Rules and RulesToInclude/RulesToExclude.'
+        Add-M365DSCEvent -Message $message -EntryType 'Error' `
+            -EventID 1 -Source $($MyInvocation.MyCommand.Source)
+        throw $message
+    }
+
+    $ConnectionMode = New-M365DSCConnection -Workload 'PowerPlatforms' `
+        -InboundParameters $PSBoundParameters
+
+    $ConnectionMode = New-M365DSCConnection -Workload 'MicrosoftGraph' `
+        -InboundParameters $PSBoundParameters
+
+    #Ensure the proper dependencies are installed in the current environment.
+    Confirm-M365DSCDependencies
+
+    #region Telemetry
+    $ResourceName = $MyInvocation.MyCommand.ModuleName -replace "MSFT_", ""
+    $CommandName = $MyInvocation.MyCommand
+    $data = Format-M365DSCTelemetryParameters -ResourceName $ResourceName `
+        -CommandName $CommandName `
+        -Parameters $PSBoundParameters
+    Add-M365DSCTelemetryEvent -Data $data
+    #endregion
+
+    try
+    {
+        $tenantid = (Get-MgContext).TenantId
+
+        $tenantIsolationPolicy = Get-PowerAppTenantIsolationPolicy -TenantId $tenantid
+
+        [Array]$allowedTenants = $tenantIsolationPolicy.properties.allowedTenants | ForEach-Object {
+            $directions = $_.direction
+            if ($directions.inbound -eq $true)
+            {
+                if ($directions.outbound -eq $true)
+                {
+                    $direction = 'Both'
+                }
+                else
+                {
+                    $direction = 'Inbound'
+                }
+            }
+            elseif ($directions.outbound -eq $true)
+            {
+                $direction = 'Outbound'
+            }
+            else
+            {
+                $direction = 'unknown'
```

Comment:
Shouldn't we throw an error here instead since unknown is not a valid value?

Reply:
Throwing an error will cause the Get/Test to fail and the Set method to never be called. By using this option, the Test will fail and the Set method will be called.